app/src/main/java/app/gamenative/statsgen/VdfParser.kt

```
@@ -0,0 +1,105 @@
+    fun binaryLoads(data: ByteArray): Map<String, Any> {
+        val stream = DataInputStream(ByteArrayInputStream(data))
+        return parseVdfData(stream)
+    }
+
+    private fun parseVdfData(stream: DataInputStream): Map<String, Any> {
+        val result = mutableMapOf<String, Any>()
+
+        while (stream.available() > 0) {
+            val dataType = try {
+                stream.readByte()
+            } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)